moabb/evaluations/evaluations.py

```diff
@@ -35,6 +43,36 @@
 Vector = Union[list, tuple, np.ndarray]
 
 
+def objective(trial, X, y, clf, scorer, epochs, random_state):
+    learning_rate = trial.suggest_float("learning_rate", 1e-4, 1e-2, log=True)
+    weight_decay = trial.suggest_float("weight_decay", 1e-10, 1e-3, log=True)
+    drop_rate = trial.suggest_float("drop_rate", 0.3, 0.9)
+
+    pre_model, model = create_deep_model(
+        clf, learning_rate, weight_decay, drop_rate, epochs=epochs
+    )
+    n_epochs = list(range(len(y)))
+    try:
+        idx_X_train, idx_X_val, y_train, y_val = train_test_split(
+            n_epochs, y, test_size=0.2, stratify=True, random_state=random_state
+        )
+    except Exception as e:
+        print(e)
+        idx_X_train, idx_X_val, y_train, y_val = train_test_split(
+            n_epochs, y, test_size=0.2, random_state=random_state
+        )
+
+    X_train = X[idx_X_train]
+    X_val = X[idx_X_val]
+    pre_model.fit(X=X_train, y=y_train)
+    X_val = pre_model.transform(X_val)
```

Comment:
Oh I see, you need this for the `deep_validation_data` only. So you need to name it differently, otherwise this will break with the scorer which will have applied the `pre_model` twice.

Reply:
True!!